pkg/blocks/fetchuri.go

```diff
@@ -0,0 +1,309 @@
+/*
+Copyright © 2023-present, Meta Platforms, Inc. and affiliates
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+package blocks
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"net/http"
+	"net/url"
+	"os"
+
+	"github.com/facebookincubator/ttpforge/pkg/logging"
+	"github.com/spf13/afero"
+	"go.uber.org/zap"
+	"gopkg.in/yaml.v3"
+)
+
+// FetchURIStep represents a step in a process that consists of a main action,
+// a cleanup action, and additional metadata.
+type FetchURIStep struct {
+	*Act        `yaml:",inline"`
+	FetchURI    string     `yaml:"fetch_uri,omitempty"`
+	Retries     string     `yaml:"retries,omitempty"`
+	Location    string     `yaml:"location,omitempty"`
```

Comment:
We might want to name this `write_to_file:` or something more explicit since in the context of fetching a URI the string `Location` makes me think of the `Location` HTTP header